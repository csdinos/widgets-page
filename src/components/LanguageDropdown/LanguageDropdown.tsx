import {Button, Container, FormControl, Grid, InputLabel, MenuItem, Select, FormHelperText} from '@material-ui/core'
import {useState, ChangeEvent} from 'react'
import {useHistory} from 'react-router-dom'
import {useDispatch, useSelector} from 'react-redux'
import * as actions from '../../actions'
import {State} from '../../types'

export const LanguageDropdown = () => {
  const language = useSelector((state: State) => state.newWidget.language)
  const dispatch = useDispatch()
  const [error, setError] = useState(false)
  const [errorMessage, setErrorMessage] = useState('')
  const history = useHistory()

  const handleChange = (event: ChangeEvent<{ value: unknown }>) => {
    dispatch(actions.setNewWidgetLanguage(event.target.value))
  }

  const onNext = () => {
    if (!language || !language.length) {
      setError(true)
      setErrorMessage('Cannot be empty')
    } else {
      history.push('/create/name')
    }
  }

  return (
    <Container maxWidth="xl">
      <Grid container
            direction="row"
            alignItems="center"
            justify="center"
            spacing={2}
      >
        <Grid item xs={12}>
          <FormControl variant="outlined" fullWidth error={error} id="language-selector"
                       data-testid="language-form-control">
            <InputLabel> Language </InputLabel>
            <Select
              fullWidth
              value={language}
              onChange={handleChange}
              label="Language"
              data-testid="language-select"
            >
              <MenuItem className="language-selector-item" value={'Greek'}>Greek</MenuItem>
              <MenuItem className="language-selector-item" value={'German'}>German</MenuItem>
              <MenuItem className="language-selector-item" value={'English'}>English</MenuItem>
              <MenuItem className="language-selector-item" value={'Polish'}>Polish</MenuItem>
            </Select>
            <FormHelperText id="error-message" hidden={!error}>{errorMessage}</FormHelperText>
          </FormControl>
        </Grid>
        <Grid item xs={12}>
          <Button
            id="next-btn"
            fullWidth
            variant="contained"
            color="primary"
            onClick={onNext}
          >Next</Button>
        </Grid>
      </Grid>
    </Container>
  )
}