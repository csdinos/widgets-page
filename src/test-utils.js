import React from 'react'
import {render as rtlRender} from '@testing-library/react'
import {createStore} from 'redux'
import {Provider} from 'react-redux'
import {reducer} from './reducers'

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({children}) {
    return <Provider store={store}>{children}</Provider>
  }

  return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
}

export const initialState = {
  widgets: [],
  newWidget: {
    id: -1,
    name: '',
    language: '',
  },
  deleteModalId: null,
  hasLoaded: false,
  shouldUpdateStorage: false
}

export * from '@testing-library/react'
export {render}