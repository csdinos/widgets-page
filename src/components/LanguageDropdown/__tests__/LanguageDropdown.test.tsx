import React from 'react'
import {render} from '../../../test-utils'
import {LanguageDropdown} from '../LanguageDropdown'

describe('<LanguageDropdown />', () => {
  test('should display an empty language selector with a next button', async () => {
    const {asFragment} = render(<LanguageDropdown/>)
    expect(asFragment()).toMatchSnapshot()
  })
})