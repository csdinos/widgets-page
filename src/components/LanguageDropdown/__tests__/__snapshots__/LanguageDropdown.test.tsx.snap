// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageDropdown /> should display an empty language selector with a next button 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
          data-testid="language-form-control"
          id="language-selector"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
             Language 
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            data-testid="language-select"
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <span>
                â€‹
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Language
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-contained"
            hidden=""
            id="error-message"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          id="next-btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Next
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
